master/internal/api_workspace.go

```
@@ -551,11 +558,16 @@ func (a *apiServer) DeleteWorkspace(
 		return nil, err
 	}
 
+	modelsExist, _ := a.workspaceHasModels(ctx, req.Id)
+	if modelsExist {
+		return nil, fmt.Errorf("workspace (%d) contains models; move or delete models first", req.Id)
+	}
+
 	holder := &workspacev1.Workspace{}
-	err = a.m.db.QueryProto("deletable_workspace", holder, req.Id)
+	// TODO(kristine): update workspace state in transaction with template delete
+	_ = a.m.db.QueryProto("deletable_workspace", holder, req.Id)
```

same comment as above about ignoring errors